### ISSUE TITLE
Image in iframe on dashboard does not refresh

### ISSUE THREAD

**Issue by marhoy** [home-assistant/appdaemon#181]
I have created a dashboard with a meteogram (weather forecast) inside an iframe. The image on the web-server is recreated every 10 minutes.

The image is displayed in the dashboard, but it is not refreshed automatically. I have to reload the dashboard in order to refresh the image. I've tested with Chrome and Safari.

I've tried different hacks, using img_list or url_list, and repeating the same url two times. No luck so far. Any idea what I am doing wrong? Is it a bug?

I'm running the latest hass.io installation (HA 0.60 and appdaemon add-on 0.3.0, the log reports AppDaemon Version 2.1.12) on an rpi3.

The relevant configuration-entries looks like this:

```
title: Gangen                
widget_dimensions: [128, 115]
widget_margins: [4, 4]                           
columns: 6                                       

meteogram:
  widget_type: iframe
  refresh: 60        
  img_list:          
    - http://www.pvv.ntnu.no/~gunde/meteogram.png

layout:                                          
  - clock, spacer, spacer, meteogram(3x2)        
```

**Comment by acockburn** [comment]
This has been discussed on the forum before I believe - the problem is that it is a browser caching issue and after much investigation we didn't find a way to make it refresh reliably.

**Comment by marhoy** [comment]
I suspected it was client side caching, so I have done a lot to prevent that. But still the image does not refresh.

- On the server-side, I have speficied a max-age of 0, so "wget -S url" reports:
```Cache-Control: max-age=0, public```

- On the client side, I'm starting Chrome with ` --disk-cache-dir=/dev/null -–disable-application-cache -–media-cache-size=1 --incognito --kiosk`.

- I have also tried to enable the "Disable cache" button in Chrome (when in developer mode), but this doesn't help either.

Is the refresh-code still active in the dashboard, or have you removed it? The above should be enough to prevent client side caching.

**Comment by christian-ek** [comment]
I had this issue. To solve it i made a custom_widget with some modifications of the baseiframe.js code. It is already implementet in the entity_picture code but not in the img_list code. So I just copied it.


```
# Starting at row 33 in baseiframe.js

        else if ("img_list" in self.parameters)
        {
            var url = self.parameters.img_list[self.index]
            url = url + "?time=" + Math.floor((new Date).getTime()/1000);
            self.set_field(self, "img_src", url);
            size = self.parameters.img_list.length
        }

```

**Comment by acockburn** [comment]
The code hasn't changed so should still be working but I'll double check. I'll also look at implementing Christian's solution as well.

**Comment by marhoy** [comment]
Thanks @christian-ek!

I can confirm that Christians solution works in combination with --incognito in Chrome :)

**Comment by acockburn** [comment]
OK, I have added that fix - it will be in 3.0. Closing the issue.
